src/js/first-run/first-run.js

```diff
@@ -1,7 +1,35 @@
 "use strict";
 
+function generatePKCECodeVerifierValue() {
+  /*
+   * A code verifier is a random string using characters of
+   * [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~",
+   * with a minimum length of 43 characters and a maximum length of 128 characters.
+   */
+  var text = "";
+  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+
+  for (var i = 0; i < Math.random() * (128 - 43) + 43; i++) {
+    text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+
+  return text;
+}
+
+
+async function pkceCodeChallengeB64(pkceCodeVerifierValue){
+  const pkceCodeChallengeDigest = await crypto.subtle.digest('SHA-256',
+    new TextEncoder().encode(pkceCodeVerifierValue)
+  );
+  const b64EncodedChallenge = btoa(String.fromCharCode(...new Uint8Array(pkceCodeChallengeDigest)))
+  return b64EncodedChallenge.replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '');
+}
+
+
 document.addEventListener("DOMContentLoaded", async () => {
   const { relaySiteOrigin } = await browser.storage.local.get("relaySiteOrigin");
+  const fxaAuthBaseUrl = 'https://oauth.stage.mozaws.net/v1';
+  const clientID = '0d134b18b51f173e';
```

Comment:
Suggestion (non-blocking): Set this as a variable at the top of the file. 

Note: I know we don't have an actual .ENV file setup, but I like to surface these where we can. 

```suggestion
  const FXA_CLIENT_ID = '0d134b18b51f173e';
```